com.unity.netcode.gameobjects/Runtime/Serialization/FastBufferReaderExtensions.cs

```diff
@@ -0,0 +1,327 @@
+
+using System;
+using UnityEngine;
+
+namespace Unity.Netcode
+{
+    public static class FastBufferReaderExtensions
+    {
+
+        /// <summary>
+        /// Reads a boxed object in a standard format
+        /// Named differently from other ReadValue methods to avoid accidental boxing
+        /// </summary>
+        /// <param name="reader">The reader to use to read the value</param>
+        /// <param name="value">The object to read</param>
+        /// <param name="type">The type to be read</param>
+        /// <param name="isNullable">
+        /// If true, reads a byte indicating whether or not the object is null.
+        /// Should match the way the object was written.
+        /// </param>
+        public static void ReadObject(this ref FastBufferReader reader, out object value, Type type, bool isNullable = false)
```

Comment:
let's kill these NO, NB, GO things in favor of #1173 🔪

Reply:
I'm not sure why #1173 is better here... it does the same thing, but requires the user to construct additional types every time they want to send. My assumption had been that #1173 was due to the fact that NO/NB/GO weren't working well with ILPP, but that's no longer the case, so... why give the users a more difficult to use interface when we can support a clean and intuitive one? (Just my opinion here of course...)

Reply:
nope, that was a design decision not a workaround due to technical limitation. we can kick-off that thread internally if you'd like but we pretty much aligned on this a while ago — I'm not sure whether you have been there or not, can reconcile convos if you want.

Reply:
Done.